Integrate changes from microbit-foundation 1.0.0 tag

microbit-sam opened the pull request:
Replacing https://github.com/bbcmicrobit/PythonEditor/pull/82

---

Bring in changes from the Microbit Educational Foundation fork

microbit-carlos commented on VERSION:
```diff
@@ -0,0 +1 @@
+1.0.0
```
If we need to update this on this release, can we also add it to the RELEASE_TODO.rst file?

microbit-carlos commented on VERSION:
```diff
@@ -0,0 +1 @@
+1.0.0
```
From the previous PR:

> Btw, I am not sure why this file is here and what it does, do you have more context @microbit-sam ?

microbit-sam replied:
I think it's only used by the build/deploy scripts so maybe it should be removed from this PR

microbit-carlos replied:
This still means that on python-editor there are two different places we need change the version number, any way it could be merged into a single source of truth? (After syncing repos)

microbit-carlos replied:
Yep absolutely. Could you create a new ticket in the python-editor repo to not forget about this?

microbit-carlos commented:
Comment from previous PR:

> Can't comment on the editor.html file in the code view, so I'll add the comment here.

> Looks like the version string is different on this branch, if your starting point was bbcmicrobit/PythonEditor master, then it means something has changed it from '0.1.0' to '0.0.0':
https://github.com/bbcmicrobit/PythonEditor/blob/5a0e321feca68046d9e15ac864398a559baf05eb/editor.html#L72-L73

> Both bbcmicrobit/PythonEditor@master and foundation/python-editor@v1.0.0 have that string as '0.1.0'. I think we should divert from both and set this to '1.0.0', as that is what we are calling this version of the editor.

microbit-carlos commented:
Comment from previous PR:
 
> Can we also update the changelog to include info about all these changes?

microbit-carlos commented:
Comment from previous PR:

> The MicroPython version in the editor.html and firmware.hex files are different, could we replace these to be the same as the python-editor@v1.0.0?
> That way the future tag in this repo will be representative of the deployed version 1.0.0 and any other cherry picked commits should apply cleanly.

microbit-sam commented:
- Removed VERSION
- Updated editor.html VERSION
- firmware.hex and editor.html firmware are the same (tested both)
- Updated CHANGELOG
- Updated README from `./show` to `./bin/show` (this is wrong in the foundation README)

microbit-carlos commented:
The MicroPython hex in the editor.html file ends without a new line:
```
:00000001FF </div>
```

Old version of DAPLink will fail to find the end of file record without it (and possible the space at the end of the record line could be a problem as well, I think DAPLink just looks for `:00000001FF\n).

microbit-carlos commented on CHANGELOG:
```diff
@@ -1,3 +1,10 @@
+1.0.0
+-----
+
+* Updated MicroPython firmware
```
Based on the PR diff, it looks like the MicroPython firmware is the same between bbcmicrobit/PythonEditor@master and python-editor@1.0.0.

microbit-sam replied:
Hmm.. I'd based that from the commit message in https://github.com/bbcmicrobit/PythonEditor/pull/84/commits/c4ef9c224ccd0346fe139e0a35d422eedc15c105

microbit-carlos replied:
Is the hex file from commit https://github.com/bbcmicrobit/PythonEditor/commit/c4ef9c224ccd0346fe139e0a35d422eedc15c105 the same as in https://github.com/bbcmicrobit/PythonEditor/blob/master/firmware.hex ?

microbit-sam replied:
Yes, firmware I've got in my clean branch is from master not the commit
I think I must have checked out the wrong file

microbit-carlos replied:
Okay, so I am looking at the v1.0.0 release: https://github.com/microbit-foundation/python-editor/tree/1.0.0

And the hex file doesn't look like it is the same than this PR:

https://github.com/microbit-foundation/python-editor/blob/1.0.0/firmware.hex
https://github.com/bbcmicrobit/PythonEditor/blob/23d73277853c621ada1343cd4fca1e5ff0d026e9/firmware.hex

Is it possible there is a missing commit to cherry pick on top of c4ef9c2?

microbit-sam replied:
I've managed to mess it up in the final commit that should have just updated the CHANGELOG and README, will amend and push!

microbit-carlos replied:
Right, so then the v1.0.0 hex file is the same as bbcmicrobit/PythonEditor@master, correct?
https://github.com/microbit-foundation/python-editor/blob/1.0.0/firmware.hex
https://github.com/bbcmicrobit/PythonEditor/blob/master/firmware.hex

microbit-sam replied:
Yeah I think this is right now, I've updated the CHANGELOG
view-source:https://python-editor-1-0-0.microbit.org/

microbit-carlos replied:
Great, thank you Sam!

microbit-carlos commented on CHANGELOG:
```diff
@@ -1,3 +1,10 @@
+
+* Updated MicroPython firmware
+* Updated styling
+* Moved show script to bin/show
```
I would also add the flags that have been deactivated by default.

microbit-carlos commented:
The PR is ready so I'll merge it, thank you Sam!

We can now start working on a PR for v1.1.0 :)
